rover_sim/src/battery_test_pub.cpp

```diff
@@ -0,0 +1,39 @@
+#include "rclcpp/rclcpp.hpp"
+#include "rover_msgs/msg/battery.hpp"
+
+class BatteryPublisher : public rclcpp::Node
+{
+  public:
+    BatteryPublisher():
+        Node("battery_publisher")
+    {
+        _pourcentage = 100;
+        _publisher = this->create_publisher<rover_msgs::msg::Battery>("/rover/auxiliary/battery", 10);
+        _timer = this->create_wall_timer(std::chrono::seconds(1), std::bind(&BatteryPublisher::publish_battery, this));
```

Comment:
link with lambda and use constant for timer perido publish

```suggestion
        _timer = this->create_wall_timer(std::chrono::milliseconds(PUB_PERIOD_MS),
                                                        [this](void)
                                                         {
                                                             this->publish_battery();
                                                         });
```